CloudWatch and X-Ray integration tutorial (#242)
kyma-project/examples: aws-cloudwatch/README.md

> # Integrate Kyma with AWS CloudWatch

Reviewer: I propose a different file structure:
- README.md
- assets
  - arch.drawio.svg
- resources
  - secret.yaml
  - aws-distro.yaml
  - logpipeline.yaml
  - tracepipeline.yaml
  - metricpipeline.yaml
- sample-app
  - deployment.yaml
  - trace-generator.yaml

## Overview 

The Kyma Telemetry module supports you in integrating with observability backends in a convenient way. This example outlines how to integrate with [AWS CloudWatch](https://aws.amazon.com/cloudwatch) as a backend. As CloudWatch is not supporting OTLP ingestion natively, it will require to deploy the [AWS Distro for OpenTelemetry](https://aws-otel.github.io) additionally. 

![overview](./integration-diagram.svg)

## Prerequisistes 

- Kyma as the target deployment environment
- AWS account with permissions to create new users and security policies
- Helm 3.x if you want to deploy open-telemetry sample application

## Installation

### Preparation

1. Export your Namespace as a variable. Replace `{NAMESPACE}` placeholder in the following command and run it:

    ```bash
    export KYMA_NS="{NAMESPACE}"
    ```
1. If you don't have created a Namespace yet, do it now:
    ```bash
    kubectl create namespace $KYMA_NS
    ```

### Create AWS IAM User

> As a first step, we need to create an IAM User assign to it specific IAM policies through which our OTEL Collector will be communicating with AWS. 

Reviewer: ```suggestion
Create an IAM User and assign to the specific IAM policies needed to let the AWS distro communicate with the AWS services
```

1. Enter the policy name and click `Create policy`

Now, we should create the policy for CloudWatch Logs:
1. Repeat the first two steps you did while creating the policy for metrics
1. Remove the `Popular services` flag, and select `CloudWatch Logs` service.
1. Now, select `CreateLogGroup`, `CreateLogStream`, `PutLogEvents` actions and click `Next`
1. Specify resource ARN for selected actions
1. Enter the policy name and click `Create policy` 

After creating the IAM Policies, we can finally create an IAM User:
1. Go to the `Users` section and click `Add user`
1. Enter the user name and click next
1. Click to the `Attach policies directly`
1. Select two new policies you added on previous steps as well as the `AWSXrayWriteOnlyAccess`
1. Click `Next` and then `Create User`
1. Open the new user
1. Go to the `Security credentials` tab and click `Create access key`
1. Select `Application running outside AWS` and then click `Next`
1. Describe the purpose of this access key and click `Create access key`
1. Now copy and save `Access key` and `Secret access key`

### Create a secret with AWS Credentials

> In order to connect OTEL Collector to AWS we need to define security credentials in the kyma system. 

Reviewer: ```suggestion
In order to connect the AWS Distro to the AWS services we need to define place the credentials of the created user into the cluster. 
```

1. In the [values.yaml](./aws-secret/values.yaml), replace the `{ACCESS_KEY}` and `{SECRET_ACCESS_KEY}` to your access keys, and `{AWS_REGION}` with the AWS region you want to use
2. Now, create the secret by using 
    ```bash

>     kubectl apply -f ./aws-secret/aws-secret.yaml

Reviewer: You don't specify a namespace. It should be the one defined as variable in the beginning

    ```

### Deploy an OTEL Collector

After creating a secret and configuering AWS, we can finally deploy an Otel Collector itself.

1. Deploy an OTEL Collector by calling 
    ```bash

>     kubectl -n $KYMA_NS apply -f ./aws-otel-collector/aws-otel.yaml

Reviewer: Is there really no source in the internet to get that content? How to maintain it?

    ```

### Create pipelines

> After deploying OTEL Collector itself, you should deploy logpipeline, tracepipeline, and metricpipeline. 

Reviewer: ```suggestion
Enable ingestion of the signals from your workloads, using the available features of the Kyma Telemetry module:
```

> 1. Deploy a logpipeline by calling 

Reviewer: ```suggestion
1. Enable a LogPipeline which is shipping container logs of all workload directly to the AWS X-Ray service, leveraging the same secret as the AWS Distro is using, bypassing the AWS Distro. As logs are not yet supported by the AWS Distro and the Kyma feature is not based on OTLP yet, the integration here is not consistent with the rest for now. Enable the LogPipeline by running:
```

    ```bash
    kubectl apply -f ./pipelines/logpipeline.yaml
    ```
1. Replace `{NAMESPACE}` and deploy a tracepipeline by calling 
    ```bash
    kubectl apply -f ./pipelines/tracepipeline.yaml
    ```
1. Replace `{NAMESPACE}` and deploy a metricpipeline by calling 
    ```bash
    kubectl apply -f ./pipelines/metricpipeline.yaml
    ```

## Verifying the results by deploying sample apps

In order to verify the results of CloudWatch and X-Ray we will deploy sample applications for each service accordingly.

### Verifying CloudWatch traces, logs, and metrics arrival 

> In order to deploy sample app which generates traces that we took from [aws-otel tutorial](https://docs.aws.amazon.com/eks/latest/userguide/sample-app.html):

Reviewer: Why we cannot deploy it directly from there? Do not duplicate the manifests without reason as we would need to maintain it
